default/config.yaml

```diff
@@ -7,9 +7,11 @@ cardsCacheCapacity: 100
 # Listen for incoming connections
 listen: false
 # Enables IPv6 and/or IPv4 protocols. Need to have at least one enabled!
+# - Use option "auto" to automatically detect support
+# - Use true or false (no qoutes) to enable or disable each protocol
 protocol:
-    ipv4: true
-    ipv6: false
+    ipv4: auto
```

Comment:
@Cohee1207 i guess there is no disadvantage when going over localhost but it shouldn't cause any problems with that configuration either, are there any modern OSes that have any ipv6 support enough to get an address but broken localhost support?

i can't really agree with ISP implementation for inbound, preferring it for outbound makes more sense, but anyone that has inbound internet traffic would have to explicitly add an AAAA record to their dns or use the ipv6 address directly.

and i also don't see a common way that hardware without ipv6 support would cause issues accessing over ipv4,
a router/firewall with bad ipv6 support, (e.g. no configurable v6 firewall) should just mean you can't access v6 over the internet or, can't get v6 at all which should just mean you can't connect over v6 outside localhost,
and a network switch without ipv6 support should just break v6 and you'd need to use v4 to connect,
but none of that stuff should need ipv6 to be disabled in ST.

Reply:
@Cohee1207 i think that's actually an implementation bug on my end, i just spent forever debugging it on linux in general,
it seems while using url formatted addresses `[::1]`, `[::]`, etc work on windows they don't on linux, i changed to using normal formatted addresses `::1`, `::`.
i also added the `ipv6Only` option to `server.listen` because depending on OS some like to bind to `0.0.0.0` if you bind to `::` which was strangely hard to find,
and now it seems to work on windows and linux
i also added a check specifically for if the system has `::1` and `127.0.0.1` too so it shouldn't try to bind to that when you're just using localhost without `--listen`